### PR TITLE
AWS NLB - controllers

### PR DESCRIPTION
**Pull request by pms1969**
Controller load balancer set to an AWS NLB

Testing
deployed/destroyed/applied all addons.

ref: #130

### CODE REVIEW

File: aws/container-linux/kubernetes/elb.tf [hunk 1]
```diff
@@ -7,37 +7,49 @@ resource "aws_route53_record" "apiserver" {
...
     healthy_threshold   = 2
-    unhealthy_threshold = 4
-    timeout             = 5
-    interval            = 6
+    unhealthy_threshold = 2
+    interval            = 10
   }
+}
+
+resource "aws_lb_listener" "apiserver-443" {
```
Comment (dghubble) [inline aws/container-linux/kubernetes/elb.tf]: Resource naming convention would be apiserver-https if you look at other places in the project.

File: aws/container-linux/kubernetes/elb.tf [hunk 2]
```diff
@@ -7,37 +7,49 @@ resource "aws_route53_record" "apiserver" {
...
-  listener {
-    lb_port           = 443
-    lb_protocol       = "tcp"
-    instance_port     = 443
-    instance_protocol = "tcp"
-  }
+resource "aws_lb" "apiserver" {
+  name               = "${var.cluster_name}-apiserver"
+  load_balancer_type = "network"
+  subnets            = ["${aws_subnet.public.*.id}"]
```
Comment (dghubble) [inline aws/container-linux/kubernetes/elb.tf]: You've removed the security group

File: aws/container-linux/kubernetes/elb.tf [hunk 3]
```diff
@@ -7,37 +7,49 @@ resource "aws_route53_record" "apiserver" {
...
+resource "aws_lb_target_group" "apiserver" {
+  name     = "${var.cluster_name}-apiserver"
+  port     = 443
+  protocol = "TCP"
+  vpc_id   = "${aws_vpc.network.id}"
 
   # Kubelet HTTP health check
   health_check {
-    target              = "SSL:443"
+    protocol            = "TCP"
```
Comment (dghubble) [inline aws/container-linux/kubernetes/elb.tf]: Intentional change?

File: aws/container-linux/kubernetes/elb.tf [hunk 4]
```diff
@@ -7,37 +7,49 @@ resource "aws_route53_record" "apiserver" {
...
     healthy_threshold   = 2
-    unhealthy_threshold = 4
-    timeout             = 5
-    interval            = 6
+    unhealthy_threshold = 2
```
Comment (dghubble) [inline aws/container-linux/kubernetes/elb.tf]: Restore the original health check values please. Let's not slip in unrelated changes.

File: aws/container-linux/kubernetes/elb.tf [hunk 5]
```diff
@@ -7,37 +7,49 @@ resource "aws_route53_record" "apiserver" {
...
-  idle_timeout                = 3600
-  connection_draining         = true
-  connection_draining_timeout = 300
+  default_action {
+    target_group_arn = "${aws_lb_target_group.apiserver.arn}"
+    type             = "forward"
+  }
 }
+
+resource "aws_lb_target_group_attachment" "apiserver" {
```
Comment (dghubble) [inline aws/container-linux/kubernetes/elb.tf]: "apiservers" or better yet "controllers" would match the established patterns

File: aws/container-linux/kubernetes/elb.tf (hunk 2 above)
Comment (dghubble) [reply aws/container-linux/kubernetes/elb.tf]: Disregard, I see network load balancer doesn't allow it.

File: aws/container-linux/kubernetes/elb.tf (hunk 3 above)
Comment (pms1969) [reply aws/container-linux/kubernetes/elb.tf]: Yes.  Required to get the healthcheck working with nlb's.

File: aws/container-linux/kubernetes/elb.tf (hunk 5 above)
Comment (pms1969) [reply aws/container-linux/kubernetes/elb.tf]: I'll provide a fix soon changing it to controllers.

File: aws/container-linux/kubernetes/elb.tf (hunk 1 above)
Comment (pms1969) [reply aws/container-linux/kubernetes/elb.tf]: I'll provide a change soon'ish.  probably tomorrow morning.

**Comment by pms1969** [comment]
In answer to your question: 
> What is the minimum version of the AWS provider you've evaluated this against?

My current setup is.
```
$ terraform version
Terraform v0.11.3
+ provider.aws v1.9.0
+ provider.ct (unversioned)
+ provider.local v1.1.0
+ provider.null v1.0.0
+ provider.template v1.0.0
+ provider.tls v1.0.1
```

**Comment by dghubble** [comment]
The new minimum version for the terraform-provider-aws plugin is 1.7.0, as established in https://github.com/poseidon/typhoon/pull/141. You won't have to worry about it after all, since that's already made its way to master.

**Comment by dghubble** [comment]
Squashed and merged with a few minor tweaks in ceb555522266000c861acda361ee0a4a349b5bf9. Thanks!
